src/main/java/org/commonjava/indy/service/httprox/util/CustomClientRequestFilter.java

```diff
@@ -17,11 +18,11 @@ public class CustomClientRequestFilter implements ClientRequestFilter
 {
 
     @Inject
-    Tokens tokens;
+    OidcClient client;
 
     @Override
     public void filter( ClientRequestContext requestContext )
     {
-        requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, "Bearer " + tokens.getToken());
+        requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, "Bearer " + client.getTokens().await().indefinitely().getAccessToken());
```

Comment:
is this a token never expired? do you need to cache it?

Reply:
This will request the token each time, since creating repo is not happened frequently, it's fine to request it each time. But sure we can consider to cache it if needed later.

Reply:
And another better way to do this is using the refresh_token, but starting from Keycloak 12 a refresh token will not be returned by default for client_credentials. 
`
Tokens tokens = currentTokens;
if (tokens.isAccessTokenExpired()) {
    // Add @Blocking method annotation if this code is used with Reactive RestClient
    tokens = client.refreshTokens(tokens.getRefreshToken()).await().indefinitely();
    currentTokens = tokens;
}
`
Details ref: https://quarkus.io/guides/security-openid-connect-client

Reply:
+1